application/src/main/java/org/togetherjava/tjbot/features/reminder/RemindRoutine.java

```diff
@@ -147,34 +134,29 @@ static ReminderRoute toPrivate(PrivateChannel channel) {
         }
     }
 
-    private void reminderFailureAction(int id, long channelId, long authorId,
-                                       CharSequence content, long guildId, int reminderFailureCounter) {
-        if (reminderFailureCounter <= 3) {
-            Instant current = Instant.now();
-            database.write(context -> context.newRecord(PENDING_REMINDERS)
-                    .setId(id)
-                    .setCreatedAt(Instant.now())
-                    .setGuildId(guildId)
-                    .setChannelId(channelId)
-                    .setAuthorId(authorId)
-                    .setRemindAt(current.plusSeconds(60))
-                    .setContent(content.toString())
-                    .setReminderFailureCounter(reminderFailureCounter + 1)
-                    .insert());
-        } else {
+    private void attemptRetryReminder(Reminder reminder) {
+        if (!(reminder.failureAttempts() <= MAX_FAILURE_RETRY)) {
             logger.warn(
                     """
-                            Failed to send a reminder with (authorID '{}'), skipping it. This can be due to a network issue, \
+                            Failed to send a reminder with (authorID '{}') and (content '{}'), skipping it. This can be due to a network issue, \
                             but also happen if the bot disconnected from the target guild and the \
-                            user has disabled DMs or has been deleted."""+ "contents: "+content,
-                    authorId);
-        }
-    }
+                            user has disabled DMs or has been deleted.""",
+                    reminder.authorId(),reminder.content());
+            return;
 
-        private Integer getReminderFailureCounter(int id){
-            return database.read(context -> context.selectFrom(PENDING_REMINDERS).where(PENDING_REMINDERS.ID.eq(id))
-                    .stream().map(PendingRemindersRecord::getReminderFailureCounter).findFirst().get()
-            );
         }
+
+        Instant current = Instant.now().plusSeconds(60);
+        database.write(context -> context.newRecord(PENDING_REMINDERS)
+                .setId(reminder.id())
+                        .setCreatedAt(reminder.createdAt())
+                .setGuildId(reminder.guildId())
+                .setChannelId(reminder.channelId())
+                .setAuthorId(reminder.authorId())
+                .setRemindAt(current)
+                .setContent(reminder.content())
+                .setReminderFailureCounter(reminder.failureAttempts()+1)
```

Comment:
to make it visually standing out better, id put this crucial line into a seperate variable beforehand:
```java
int failureAttempts = reminder.failureAttempts() + 1;

...

  .setReminderFailureCounter(failureAttempts)
```